### PR TITLE
Add gradient for inverse hyperbolic functions

### PR DESCRIPTION
**Pull request by 0urobor0s**
This commit adds gradient for the following inverse hyperbolic functions:

- Arcsinh
- Arccosh
- Arctanh

PS: Shouldn't this functions drop the c? Wikipedia has some information on this. [Notation](https://en.wikipedia.org/wiki/Inverse_hyperbolic_functions#Notation)

### CODE REVIEW

File: nx/test/nx/defn/grad_test.exs [hunk 1]
```diff
@@ -509,6 +509,29 @@ defmodule Nx.Defn.GradTest do
...
 
+  describe "inverse hyperbolic functions" do
+    defn grad_arcsinh(t), do: grad(t, Nx.arcsinh(t))
+    defn grad_arccosh(t), do: grad(t, Nx.arccosh(t))
+    defn grad_arctanh(t), do: grad(t, Nx.arctanh(t))
+
+    test "computes gradient of inverse hyperbolic functions" do
+      for _ <- @iters do
+        t = Nx.random_uniform({}, -100.0, 100.0, type: {:f, 64})
+        check_grads!(&Nx.arcsinh/1, &grad_arcsinh/1, t, eps: 0.1)
```
Comment (josevalim) [inline nx/test/nx/defn/grad_test.exs:520]: I assume a lower eps fails frequently?

**Comment by josevalim** [comment]
> PS: Shouldn't this functions drop the c? Wikipedia has some information on this. Notation

Good call. I will rename them to asin, acos, etc after I merge this. This is the notation used by Julia, Erlang, and C's math.

**Comment by josevalim** [comment]
Merging so I can do the rename, we can revisit the EPS later accordingly.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

File: nx/test/nx/defn/grad_test.exs (hunk 1 above)
Comment (0urobor0s) [reply nx/test/nx/defn/grad_test.exs:520]: I think it's fine to lower it or remove it to use the default value.

**Comment by 0urobor0s** [comment]
> > PS: Shouldn't this functions drop the c? Wikipedia has some information on this. Notation
> 
> Good call. I will rename them to asin, acos, etc after I merge this. This is the notation used by Julia, Erlang, and C's math.

Apparently in Wikipedia, following the ISO 80000-2 it should be, ar<func>h, but following Erlang or C might be easier to for developers as it reduces the things to remember.

**Pull request merged**